### ISSUE TITLE
Mad-Pascal/blibs/b_utils.pas (397,6) Error: Identifier not found 'DATE.DAY'

### ISSUE THREAD

**Issue by mrgw454** [drunkeneye/MAFIA.A8#1]
Hello,

First -- thanks for sharing your project for us Atari 8 bit fans!

I'm trying to build this from source and believe I have all the prerequisites installed.  I'm getting the following error:

```
mp title.pas -code:0c00 -o ./output/title.a65
Mad Pascal Compiler version 1.7.1 [2024/03/04] for 6502
Compiling title.pas
../../Mad-Pascal/blibs/b_utils.pas (397,6) Error: Identifier not found 'DATE.DAY'
make: *** [makefile:38: compile_title] Error 2
```
Could it have something to do with the version of the Mad-Pascal compiler I'm using?  That was built from source just recently.

Thanks for any help you might be able to provide.

**Comment by drunkeneye** [comment]
yeah, i am sorry, i used madpascal 1.7.0, not 1.7.1.
i tried 1.7.1 near the end, and got the same error.
did not quite understood why, so i refrained from updating 
and kept 1.7.0.
could you try out 1.7.0 for now?

**Comment by mrgw454** [comment]
I can certainly try an older version.  Any idea what date v1.70 was available?  I'm pulling source from git but it's hard to see the version information.  I see the official releases only go to 1.69 (and that was Oct 12, 2023).

Thanks for the quick follow-up!

**Comment by mrgw454** [comment]
Quick update -- I did a checkout of [de4e70c] and I see 'mp' shows version 1.7.0 now.

Getting a different error so I'm think you might be tied to a very specific version of Mad-Pascal.  Here's the error now (seems minor):

```
apultra ../assets/E700PAGE.gfx_EN atr.assets/E700PAGE.apl
mp title.pas -code:0c00 -o ./output/title.a65
Mad Pascal Compiler version 1.7.0 [2024/03/04] for 6502
Compiling title.pas
../../Mad-Pascal/blibs/xbios.pas (804,49) Error: Syntax error, ';' expected but ':' found
make: *** [makefile:38: compile_title] Error 2

```

**Comment by drunkeneye** [comment]
yeah, i used this version, but i think any 1.7.0 should do it for now:
Mad Pascal Compiler version 1.7.0 [2023/11/24] for 6502

yes, the xbios.pas is again an error there in the lib, just replace the ":" with ";" in the line 804.
there was also some other error with a missing label, not sure where it popped up.
if you have other errors, just write ;)
thanks again for testing :D

**Comment by mrgw454** [comment]
Ha!  I found the parent version where that : issue was fixed (e2ad5fb) and mp still shows version 1.7.0.

Got passed that error and now on to another.  I believe this is a path/library issue since my location of Mad-Pascal is located in a different relative location than your 'makefile', 'title.pas' and 'main.pas' show.  Mine would be:

`../../Mad-Pascal
`
... from the MAFIA.A8/src folder.

Here's the error:

```
14789 lines of source assembled in 6 pass
22289 bytes written to the object file
cp ./output/title.xex ./atr.assets/XAUTORUN
cp ../assets/logo.apl ./atr.assets
cp ../assets/title.xex ./atr.assets
mp main.pas -code:0c00 -o ./output/main.a65  -define:LOCATION_EN
Mad Pascal Compiler version 1.7.0 [2024/03/04] for 6502
Compiling main.pas
player.pas (87,6) Error: Can't open include file '/home/ron/source/Mad-Pascal/src/helpersgame.pas'
make: *** [makefile:46: compile_main] Error 2
```

I need to verify which variable defines where the helpersgame.pas file is to be found (even though I see it in the MAFIA.A8 src folder.

I appreciate your help with this -- but don't want to take you away from anything more urgent.  :)

-Ron

**Comment by drunkeneye** [comment]
i think that might be due to case-sensitivity of your filesystem.
since i use macos, there is no such thing.
i think you might be on linux?
in that case you must rename the files or change the sourcecode...

**Comment by tebe6502** [comment]
https://github.com/tebe6502/Mad-Pascal

compiler no longer favors the SYSTEM unit (there is also a definition of TDateTime there, but the field names are different)

**Comment by drunkeneye** [comment]
> https://github.com/tebe6502/Mad-Pascal
> 
> compiler no longer favors the SYSTEM unit (there is also a definition of TDateTime there, but the field names are different)

yes, gave it a try just now, and the error is gone. however, something else is broken with my code, after compiling it crashes when trying to load a file. will try to solve that and then move to mp 1.7.1

**Comment by mrgw454** [comment]
I appreciate the updates on this!

**Comment by drunkeneye** [comment]
i have now fixed the problem with MP 1.7.1 (compiled at 6th march), 
seemingly pointers are handled differently now (possibly more correctly).

so i will close this issue.
